src/app/pages/PerpetualPage/utils/contractUtils.tsx

```diff
@@ -257,5 +247,32 @@ export const validatePositionChange = (
     );
   }
 
+  if (amountChange !== 0 || marginChange !== 0) {
+    const targetAmount = amountChange + traderState.marginAccountPositionBC;
+
+    const requiredMargin = getRequiredMarginCollateralWithGasFees(
+      targetLeverage,
+      targetAmount,
+      perpParameters,
+      ammState,
+      traderState,
+      slippage,
+      useMetaTransactions,
+    );
+
+    if (requiredMargin > traderState.availableCashCC + availableBalance) {
+      result.valid = false;
+      result.isWarning = true;
+      result.errors.push(new Error('Order cost exceeds total balance!'));
```

Comment:
We need to use localized strings.

Reply:
The error message here is just for logging not for displaying purposes, that's what the result.errorMessages are for.